code/io/timer.cpp

```diff
@@ -104,7 +106,12 @@ std::uint64_t timer_get_nanoseconds()
 {
 	auto time = get_performance_counter();
 
-	return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
+	if (Timer_perf_counter_freq < NANOSECONDS_PER_SECOND) {
+		// This can cause overflows but I don't know a better solution...
+		return (time * NANOSECONDS_PER_SECOND) / Timer_perf_counter_freq;
```

Comment:
How about

time * (NS/GCD(NS, tpcf) )/ (tpcf/GCD(NS, tpcf)); both values of the division can be cached as they are consts.